includes/class-amp-theme-support.php

```diff
@@ -1098,10 +1082,30 @@ public static function prepare_response( $response, $args = array() ) {
 		}
 		$dom = AMP_DOM_Utils::get_dom( $response );
 
-		self::ensure_required_markup( $dom );
+		$xpath = new DOMXPath( $dom );
+
+		$head = $dom->getElementsByTagName( 'head' )->item( 0 );
+
+		if ( isset( $head ) ) {
+			// Make sure scripts from the body get moved to the head.
+			$scripts = array();
```

Comment:
Any reason to store the scripts in a var and then run two loops rather than doing the following:
``` PHP
foreach ( $xpath->query( '//body//script[ @custom-element or @custom-template ]' ) as $script ) {
	$head->appendChild( $script );
}
```

Reply:
Makes sense :)